### ISSUE TITLE
Numerical error info should be controlled by `verbose` flag

### ISSUE THREAD

**Issue by xukai92** [TuringLang/AdvancedHMC.jl#88]
I think users should be suggested to post analysis on the chain to see the numerical errors. Or somehow we warn them when there are too many?

**Comment by yebai** [comment]
Instead of always printing or disabling these messages, maybe we can improve by storing the messages (e.g. via `Logging`) and return it. This allows the user to look at these messages if things went wrong. It also avoids the clutter if there are too many numerical errors.

**Comment by xukai92** [comment]
Yes you are right. Users just need to set global logger's IO to text. We probably just need to provide a helper function for this.

**Comment by cpfiffer** [comment]
Could you just print them with an `@debug` call, which would only print if the users were running with debug mode on? Or perhaps that is too restrictive. I very rarely run with debug logging.

**Comment by xukai92** [comment]
Related PR: #297
